manual/manual/html_processing/src/process_manual.ml

```diff
@@ -0,0 +1,465 @@
+      end
+    end
+  end;
+
+  (* Add back link to "OCaml Manual" *)
+  if file <> "index.html" then begin
+    let toc_title = create_element "div" ~class_:"toc_title" in
+    let a = create_element "a" ~inner_text:"< The OCaml Manual"
+        ~attributes:["href", "index.html"] in
+    append_child toc_title a;
+    prepend_child nav toc_title
+  end;
+
+  (* Add version number *)
+  let version_text = if file = "index.html" then "Select another version"
+    else "Version " ^ version in
+  add_version_link nav version_text releases_url;
+
+  (* Create new menu *)
+  let menu = create_element "ul" ~id:"part-menu" in
+  List.iter (fun (href, title) ->
+      let a = create_element "a" ~inner_text:title ~attributes:["href", href] in
+      let li = if href = file
+        then create_element "li" ~class_:"active"
+        else create_element "li" in
+      append_child li a;
+      append_child menu li) chapters;
+  (* let body = soup $ "div.content" in *)
+  prepend_child body menu;
+
+  (* Add part_title just before the part-menu *)
+  if part_title <> "" then begin
+    let nav = create_element ~id:"part-title" "nav" ~inner_text:part_title in
+    create_element "span" ~inner_text:"☰"
+    |> prepend_child nav;
+    prepend_child body nav
+  end;
+
+  (* Add side-bar button before part_title *)
+  add_sidebar_button body;
+
+  (* Add logo *)
+  begin match soup $? "header" with
+  | None -> dbg "Warning: no <header> for %s" file
+  | Some header -> prepend_child header (logo_html "https://ocaml.org/")
+  end;
+
+  (* Move authors to the end. Versions >= 4.05 use c009. *)
+  ["span.c009"]
+  |> List.iter (fun selector ->
+      soup $? selector
+      |> Option.iter (fun authors ->
+          match leaf_text authors with
+          | None -> ()
+          | Some s ->
+              match Re.Str.(search_forward (regexp "(.+written by.+)") s 0) with
+              | exception Not_found -> ()
+              | _ ->
+                  dbg "Moving authors";
+                  delete authors;
+                  add_class "authors" authors;
+                  append_child body authors));
+
+  (* Get the list of external files linked by the current file *)
+  let xfiles = match toc with
+    | None -> []
+    | Some toc ->
+        toc $$ "li"
+        |> fold (fun list li ->
+            let rf = li $ "a" |> R.attribute "href" in
+            dbg "TOC reference = %s" rf;
+            if not (String.contains rf '#') &&
+               not (starts_with ".." rf) &&
+               not (starts_with "http" rf)
+            then begin
+              li $ "a" |> set_attribute "href" (rf ^ "#start-section");
+              rf::list
+            end else list) []
+  in
+
+  (* Bottom navigation links *)
+  update_navigation soup toc_table;
+
+  (* Add copyright *)
+  append_child body (copyright ());
+
+  (* Generate external files *)
+  List.iter (clone_structure soup toc_table) xfiles;
+
+  (* And finally save *)
+  save_to_file soup file
```

Comment:
If you commute `save_to_file` and the generation of the external files, I have the impression that you don't need to deep copy the `soup` nodes.

Reply:
excellent point. This allows indeed the removal of all clones.